### PR TITLE
Segmented Control / Switch Field

### PR DESCRIPTION
**Pull request by jeffleeismyhero**
This adds the functionality requested in the Feature Request Issue #536. It might also work for the functionality requested in issues #474 and #476.

**Field type:** `segment`
**Values:** This field takes values in the same format as select fields.

**Example JSON**
```json
[
  {
    "id": "group",
    "title": "Group",
    "sections": [
      {
        "id": "section",
        "fields": [
          {
            "id": "segment",
            "title": "Location",
            "type": "segment",
            "values":[
              {
                "id":"in_house",
                "title":"In-house",
                "default":true,
              },
              {
                "id":"remote",
                "title":"Remote",
              }
            ],
            "size": {
              "width": 40,
              "height": 1
            }
          }
        ]
      }
    ]
  }
]
```

### CODE REVIEW

**Comment by jeffleeismyhero** [comment]
Below is a screenshot of the `segment` field with default styling.

screen shot 2016-10-10 at 3 56 35 pm

This is the field with modified font styling:

screen shot 2016-10-10 at 3 54 56 pm

**Comment by 3lvis** [comment]
Merging to `switches` branch. Gonna have a look there why the extra `value` is needed.

File: Tests/JSONs/segment-field.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "title": "Group",
+    "sections": [
+      {
+        "id": "section",
+        "fields": [
+          {
+            "id": "segment",
+            "title": "My Segment",
+            "type": "segment",
+            "value": "part_1",
```
Comment (3lvis) [inline Tests/JSONs/segment-field.json]: This wasn't needed. :)

https://github.com/hyperoslo/Form/pull/545/commits/e56a5bd6aa7140ce5751e290255a0c05df0cb2f3

**Comment by 3lvis** [comment]
👏  Fantastic job, @jeffleeismyhero 👏

**Pull request merged**